### ISSUE TITLE
A build is not scheduled after a merge operation on the branch

### ISSUE THREAD

**Issue by aproxs** [Shippable/support#1132]
Very simple repro:

git repository with 2 branches: dev and master, and dev is ahead of master.

o - abcdef123 <- dev
|
o
|
o - 9876540 <- master --- origin/master --- origin/dev
...

As I expected, a build is scheduled in shippable when I do:
git checkout dev
git push

and then I'm in the following state:
o - abcdef123 <- dev --- origin/dev
|
o
|
o - 9876540 <- master --- origin/master
...

However, when I merge my changes to master and push master, a build is not scheduled.
In other words. I do:
git checkout master
git push

(new state)
o - abcdef123 <- dev --- master --- origin/master --- origin/dev
|
o
|
o - 9876540
...

But no build is scheduled.

In our case dev moves fast and when the code is ready we merge the changes into master, which should trigger an automatic deployment, but because of this issue our deployments never get to run.

**Comment by rageshkrishna** [comment]
@aproxs I can't seem to find a Shippable account with your Github username, so can you please tell me if you're using BitBucket?

BitBucket has a bug where their webhook payload for fast-forward merges does not include the name of the branch that was updated. We're unable to trigger a build because of this. You can look at https://bitbucket.org/site/master/issue/7775/post-service-does-not-provide-useful for more details from BitBucket on this issue. There are some workarounds listed in that issue that may be helpful.

If, however, you are using Github please send me your project ID so I can take a closer look. Thanks!

**Comment by aproxs** [comment]
Yes, sorry for not posting the info.

My account is luis@latonaventures.com and yes. I'm using bitbucket.

I'll take a look at the workaround. Thanks!!

Luis

**Comment by manishas** [comment]
@aproxs i am closing this issue since there isn't much we can do at our end :(

please open a new issue if you need assistance.
